src/category_theory/abelian/right_derived.lean

```diff
@@ -0,0 +1,286 @@
+  {X : C} (P : InjectiveResolution X) : right_derived_zero_to_self_app F P ≫
+  right_derived_zero_to_self_app_inv F P = 𝟙 _ := -- sorry
+begin
+  dsimp [right_derived_zero_to_self_app, right_derived_zero_to_self_app_inv],
+  rw [map_id, category.id_comp, category.assoc],
+  refine (iso.eq_inv_comp _).1 _,
+  rw [← category.assoc, ← category.assoc],
+  refine (iso.comp_inv_eq _).2 _,
+  rw [category.comp_id, iso.inv_hom_id, category.assoc, category.assoc, iso.hom_comp_eq_id],
+  ext,
+  simp [category.assoc, homology.lift_ι_assoc,
+    cokernel.π_desc, homology.ι, iso.inv_hom_id, category.id_comp, category.comp_id],
+  nth_rewrite 1 [← category.id_comp (kernel.ι _)],
+  rw [← category.assoc, ← category.assoc, ← category.assoc],
+  congr' 1,
+  dsimp [homology.lift],
+  rw [category.assoc, category.assoc, category.assoc, iso.inv_hom_id, category.comp_id],
+  ext,
+  simp [limits.equalizer_as_kernel, category.assoc, limits.kernel.lift_ι_assoc,
+    limits.kernel.lift_ι, category.comp_id],
+  nth_rewrite 1 [← category.comp_id (kernel.ι _)],
+  congr' 1,
+  ext,
+  simp,
+end
+
+lemma right_derived_zero_to_self_app_inv_comp [enough_injectives C] [preserves_finite_limits F]
+  {X : C} (P : InjectiveResolution X) : right_derived_zero_to_self_app_inv F P ≫
+  right_derived_zero_to_self_app F P = 𝟙 _ :=
+begin
+  dsimp [right_derived_zero_to_self_app, right_derived_zero_to_self_app_inv],
+  rw [map_id, category.id_comp, ← category.assoc _ (F.right_derived_obj_iso 0 P).hom,
+    category.assoc _ _ (F.right_derived_obj_iso 0 P).hom, iso.inv_hom_id, category.comp_id,
+    ← category.assoc, ← category.assoc, is_iso.comp_inv_eq, category.id_comp],
+  ext,
+  simp only [limits.kernel.lift_ι_assoc, category.assoc, limits.kernel.lift_ι, homology.lift],
+  rw [← category.assoc, ← category.assoc, category.assoc _ _ (homology_iso_kernel_desc _ _ _).hom,
+    iso.inv_hom_id, category.comp_id],
+  simp,
+end
+
+/-- Given `P : InjectiveResolution X`, the isomorphism `(F.right_derived 0).obj X ≅ F.obj X` if
+`preserves_finite_§limits F`. -/
+def right_derived_zero_to_self_app_iso [enough_injectives C] [preserves_finite_limits F]
+  {X : C} (P : InjectiveResolution X) : (F.right_derived 0).obj X ≅ F.obj X :=
+{ hom := right_derived_zero_to_self_app _ P,
+  inv := right_derived_zero_to_self_app_inv _ P,
+  hom_inv_id' := right_derived_zero_to_self_app_comp_inv _ P,
+  inv_hom_id' := right_derived_zero_to_self_app_inv_comp _ P }
+
+/-- Given `P : InjectiveResolution X` and `Q : InjectiveResolution Y` and a morphism `f : X ⟶ Y`,
+naturality of the square given by `right_derived_zero_to_self_natural`. -/
+lemma right_derived_zero_to_self_natural [enough_injectives C] [preserves_finite_limits F]
+  {X : C} {Y : C} (f : X ⟶ Y)
+  (P : InjectiveResolution X) (Q : InjectiveResolution Y) :
+  (F.right_derived 0).map f ≫ right_derived_zero_to_self_app F Q =
+  right_derived_zero_to_self_app F P ≫ F.map f :=
+begin
+  sorry
+end
```

Comment:
@riccardobrasca , I couldn't figure out how to prove this statement (dual to `left_derived_zero_to_self_natural`), can you help me please

Reply:
never mind! I figured it out!